frontend/src/javascript/utils/downloadUtils.js

```diff
@@ -0,0 +1,68 @@
+import * as pdfMake from "pdfmake/build/pdfmake";
+import { useProjectStore } from "@/stores/project";
+import { getCurrentThemeFromStorage } from "@/javascript/storage/themeStorage";
+
+export function exportToPDF(stage) {
+    const projectStore = useProjectStore();
+    var documentDefinition = getBlankPDFDocumentDefinition();
+    const drawableWidth = documentDefinition.pageSize.width - 2 * documentDefinition.pageMargins[0];
+    const drawableHeight = documentDefinition.pageSize.height - 2 * documentDefinition.pageMargins[1];
+
+    for (let frame of projectStore.allFrames) {
+        stage.draw(projectStore.sceneObjects, frame, projectStore.updateSceneObjectPosition);
+        documentDefinition.images["frame_" + frame.id] = stage.getBase64Image();
+        documentDefinition.content.push({
+            image: "frame_" + frame.id,
+            fit: [drawableWidth, drawableHeight],
+            pageBreak: "after",
+        });
+    }
+    stage.draw(projectStore.sceneObjects, projectStore.currentFrame, projectStore.updateSceneObjectPosition);
+
+    if (documentDefinition.content.length > 0) {
+        documentDefinition.content[documentDefinition.content.length - 1].pageBreak = undefined;
+    }
+
+    pdfMake.createPdf(documentDefinition).download("algodebug.pdf");
```

Comment:
Jak Ci się chce to proponuję dodać od razu feature'a żeby dodawać nazwę projektu do nazwy pliku, np. algodebug-<nazwa_projektu>.pdf, żeby w przypadku pobierania wielu pdfów można było wygodniej rozróżnić. Do zastanowienia czy nie trzeba dorobić z tego powodu jakieś walidacji dla nazw projektu.